### ISSUE TITLE
PSR2.ControlStructures.SwitchDeclaration.BreakIndent false positive when case keyword is not indented

### ISSUE THREAD

**Issue by umherirrender** [squizlabs/PHP_CodeSniffer#3157]
**Describe the bug**
The `SwitchDeclaration` sniff fails to determine the correct indent for the `break` statement when the `case` is on the same level as `switch`

**Code sample**
```php
				switch ( $tokens[$next]['code'] ) {
				case T_OPEN_CURLY_BRACKET:
				case T_OPEN_SQUARE_BRACKET:
				case T_OPEN_PARENTHESIS:
					if ( isset( $tokens[$next]['parenthesis_closer'] ) ) {
						// jump to closing parenthesis to ignore commas between opener and closer
						$next = $tokens[$next]['parenthesis_closer'];
					}
					break; # <!--- ERROR | [x] Terminating statement must be indented to the same level as the CASE body
				case T_COMMA:
					$argCount++;
					break; # <!--- ERROR | [x] Terminating statement must be indented to the same level as the CASE body
				}
```
**Expected behavior**
The autofix adds a indent of 4 spaces to the break. Expected is no error reporting and no autofix *or* the cases are also reported and autofixed.

**Versions (please complete the following information):**
 - PHPCS: 3.5.8

**Comment by gsherwood** [comment]
The sniff is making an assumption that the CASE keywords are indented 4 spaces from the SWITCH keyword. So it is then enforcing the rule that the breaking statements must be indented 8 spaces from the SWITCH keyword. Obviously, the error message is confusing here due to this assumption.

When taken as a whole, the PSR2 standard does reformat this code correctly because the ScopeIndent sniff is the one that ensures CASE statements are indented 4 spaces from the SWITCH. When taken alone, this sniff will not enforce that part of the standard.

So there are 2 ways forward here:
1. Have the sniff enforce that CASE statements must be indented 4 spaces from the SWITCH keyword, which will make the sniff more valuable by itself, and fix the error message. This does mean a double-up of checking CASE indents though as the ScopeIndent sniff is going to do it regardless.
2. Hide the error message when it is confusing and let ScopeIndent report an error for the CASE statement.

I'm learning towards option 2 at the moment, but will look into it more.

**Comment by gsherwood** [comment]
I played around with option 1 but I wasn't confident it would catch all cases, so I implemented option 2 instead.

Thanks for the bug report.
